core/dbt/task/run_operation.py

```diff
@@ -66,8 +67,17 @@ def run(self) -> RunResultsArtifact:
         end = datetime.utcnow()
 
         package_name, macro_name = self._get_macro_parts()
-        fqn = [NodeType.Operation, package_name, macro_name]
-        unique_id = ".".join(fqn)
+        macro = self.manifest.find_macro_by_name(  # type: ignore[union-attr]
+            macro_name, self.config.project_name, package_name
+        )
+
+        try:
+            unique_id = macro.unique_id  # type: ignore[union-attr]
+            fqn = unique_id.split(".")
+        except AttributeError:
+            raise DbtRuntimeError(
+                f"dbt could not find a macro with the name '{macro_name}' in any package"
+            )
```

Comment:
This section could be refactored as following to avoid the #type: ignore workarounds and make it more clear what's happening with the `macro.unique_id` access. Catching the error on the `None` value of the returned macro feels like an indirect way of checking whether the macro was found or not. 

Additionally, we could avoid making a breaking change to the `RunResult` unique id (from `operation.` to `macro.`) if we use the attributes on the resolved macro: 

```python
        macro = (
            self.manifest.find_macro_by_name(macro_name, self.config.project_name, package_name)
            if self.manifest
            else None
        )

        if macro:
            fqn = [NodeType.Operation, macro.package_name, macro.name]
            unique_id = macro.unique_id
        else:
            raise DbtRuntimeError(
                f"dbt could not find a macro with the name '{macro_name}' in any package"
            )
```

Reply:
> Additionally, we could avoid making a breaking change to the `RunResult` unique id (from `operation.` to `macro.`)

FWIW - I believe this is new in v1.6 (https://github.com/dbt-labs/dbt-core/pull/7655), so it's not a breaking change for existing functionality in prior versions. Whether this node is better considered a `macro` or an `operation` is a fair question. IIRC, we use the `operation` node type for `on-run-*` hooks ("project-level" hooks). Whereas the resource run by `run-operation` is really just a `macro`... but of course, the command is named `run-operation`.